Implements #343 - Creating a tabbed admin interface

m-furman opened the pull request:
Resolves #343.

Split the admin interface into separate tabs. The map visualization is loaded in the default active tab, while all the graphs/charts are only loaded once the 'Analytics' tab is clicked for the first time.

I also addressed the comment that Manaswi left in my pull request for #386.

manaswis commented:
Looks good! Can you add screenshots of the interface for others to see? Also, could you test and compare the loading times between the newer and older version?

*Suggested Improvement*:
Adjust the table width size of the activities and coverage table to spread out the data in the tables a bit more and cover the whole page width.
<img width="1280" alt="screen shot 2017-02-12 at 10 05 46 pm" src="https://cloud.githubusercontent.com/assets/2873216/22889145/1f54cb38-f1d6-11e6-81a1-71ae62d10727.png">

m-furman commented:
Here are the screenshots! I'm going to try to compare the loading times of the pages soon.

Positioning of the  tabs relative to the Project Sidewalk logo:
<img src="http://drive.google.com/uc?export=view&id=0B687BEjGtgFVa3Nnb1hQSlY5dWc"  />

Tab 1 content:
<img src="http://drive.google.com/uc?export=view&id=0B687BEjGtgFVaU9DTElFSFJhRXc"  />
Tab 2 content:
<img src="http://drive.google.com/uc?export=view&id=0B687BEjGtgFVejRFa0hPQUlFVFU"  />
Tab 3 content:
<img src="http://drive.google.com/uc?export=view&id=0B687BEjGtgFVaFE4bWN3RHJsTVk"  />
Tab 4 content:
<img src="http://drive.google.com/uc?export=view&id=0B687BEjGtgFVZnlLVUdTVjFLWVE"  />
Tab 5 content:
<img src="http://drive.google.com/uc?export=view&id=0B687BEjGtgFVeWgwSkZfajZud2c"  />

m-furman commented:
Also, would this be a better layout for the activities and coverage tables?
<img src="http://drive.google.com/uc?export=view&id=0B687BEjGtgFVT2plN3duOXlnVzQ"  />

manaswis commented:
Thanks for adding the screenshots. Yes, the tables look good.

m-furman commented:
So it looks like the loading time of the new version is about 5000 - 6000 milliseconds, while the loading time of the old version is about 9000 - 10000 milliseconds. I can provide more exact statistics if necessary.

manaswis commented:
Users and Anonymous Users could be in a single tab "Users" and the headings for the tables under the tab would be Registered Users and Anonymous Users.

jonfroehlich commented:
I agree. Because I am typically very interested in comparing the two groups.

On Tue, Feb 14, 2017 at 2:00 PM, Manaswi Saha <notifications@github.com>
wrote:

> Users and Anonymous Users could be in a single tab "Users" and the
> headings for the tables under the tab would be Registered Users and
> Anonymous Users.
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/ProjectSidewalk/SidewalkWebpage/pull/515#issuecomment-279801494>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ABi-9Un4B8svLF0TJ-gLyBGc0V4vjTmpks5rcfnNgaJpZM4L86w3>
> .
>



-- 
Jon Froehlich
Assistant Professor
Computer Science
University of Maryland, College Park
http://www.cs.umd.edu/~jonf/
@jonfroehlich <https://twitter.com/jonfroehlich> - Twitter

m-furman commented:
Thanks for the feedback!
I just made the change:
<img src="http://drive.google.com/uc?export=view&id=0B687BEjGtgFVbG1rdHh0Nzc1M2M"  />

manaswis commented on app/views/admin/index.scala.html:
```diff
@@ -11,385 +11,426 @@
+                            </tr>
+                            <tr style="font-size: 90%;">
+                                <td style="padding-left: 30px">Curb Ramp
+                                </th>
```
The closing tags are incorrect in this sub-section of "Activities".

<td>....</th>

manaswis replied:
6 of them starting from this one

manaswis commented:
@m-furman I reviewed your code. There are some minor changes (listed above). Please update the PR with those changes. I'll merge it with develop after that.

m-furman commented:
@manaswi I addressed the comments you left above as well as moved the map into a separate tab as you requested in your e-mail. The map query should only run once when you open the tab.

manaswis commented:
Looks good! Thanks @m-furman.
